app/Http/Request/StatementRequest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Http\Request;
+
+use Anik\Form\FormRequest;
+use App\Helpers\ResponseInterface;
+
+class StatementRequest extends FormRequest
+{
+    public function __construct(ResponseInterface $resProvider)
+    {
+        $this->resProvider = $resProvider;
+    }
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    protected function authorize(): bool
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    protected function rules(): array
+    {
+        if ($this->method() == 'GET') {
+            return [
+                'topic_num' => 'required',
+                'camp_num' => 'required',
+                'as_of' => 'in:default,review,bydate',
+                'as_of_date' => 'required_if:as_of,bydate'
+            ];
+        }
+    }
+
+    /**
+     * Get  validation messages that apply to the request.
+     *
+     * @return array
+     */
+    protected function messages(): array
+    {
+        return [
+            'as_of.in'=>"Please enter a valid value (default,review,bydate) or leave it empty",
```

Comment:
ValidationMessages.php and ValidationRules are the classes that we are using to validation messages and rules.